spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/secretmanager/SecretManagerPropertySource.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2017-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.secretmanager;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.google.cloud.secretmanager.v1beta1.AccessSecretVersionResponse;
+import com.google.cloud.secretmanager.v1beta1.ProjectName;
+import com.google.cloud.secretmanager.v1beta1.Secret;
+import com.google.cloud.secretmanager.v1beta1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1beta1.SecretManagerServiceClient.ListSecretsPagedResponse;
+import com.google.cloud.secretmanager.v1beta1.SecretVersionName;
+
+import org.springframework.cloud.gcp.core.GcpProjectIdProvider;
+import org.springframework.core.env.EnumerablePropertySource;
+
+public class SecretManagerPropertySource extends EnumerablePropertySource<SecretManagerServiceClient> {
+
+	private static final String SECRETS_NAMESPACE = "spring-cloud-gcp.secrets.";
+
+	private static final String LATEST_VERSION_STRING = "latest";
+
+	private final Map<String, Object> properties;
+
+	private final String[] propertyNames;
+
+	public SecretManagerPropertySource(
+			String propertySourceName,
+			SecretManagerServiceClient client,
+			GcpProjectIdProvider projectIdProvider) {
+
+		super(propertySourceName, client);
+
+		Map<String, Object> propertiesMap = initializePropertiesMap(client, projectIdProvider.getProjectId());
```

Comment:
You're loading properties in the constructor. Would it make sense to allow users to reload the properties by providing a public "refresh" method to do it?

Reply:
More generally, test behavior of a`@RefreshScope` property in the sample. Ideally, triggering a refresh event should get those properties to expire.

Reply:
Regarding the refresh functionality, can I verify this in a separate PR? This would be something to verify through the addition of the sample app and a Properties class.